### PR TITLE
fix: make title field readonly

### LINKED ISSUE #33806: Can't rename Lead
### Information about bug

Creating lead then save it. Then rename it, won't rename it.

### Module

CRM

### Version

ERPNext: v14.12.1 (version-14)
Frappe Framework: v14.22.0 (version-14)

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
No error, but appears a success message and the rename is not done.
```

### PR DESCRIPTION
**Pull request by dj12djdjs**
closes: https://github.com/frappe/erpnext/issues/33806#issuecomment-1402804212

Make the title field read_only so when the user clicks the title, the rename dialog won't appear.

### CODE REVIEW

**Comment by dj12djdjs** [comment]
please backport version-14

File: erpnext/crm/doctype/lead/lead.json [hunk 1]
```diff
@@ -514,11 +515,10 @@
  "idx": 5,
  "image_field": "image",
  "links": [],
- "modified": "2022-10-13 12:42:04.277879",
+ "modified": "2023-01-24 18:20:05.044791",
  "modified_by": "Administrator",
  "module": "CRM",
  "name": "Lead",
- "name_case": "Title Case",
```
Comment (deepeshgarg007) [inline erpnext/crm/doctype/lead/lead.json:521]: @dj12djdjs Why is this removed?

Comment (dj12djdjs) [reply erpnext/crm/doctype/lead/lead.json:521]: @deepeshgarg007 I'm not sure. I only checked Read Only on the title field. Clicked Save.

This actually happened last on my last PR changing a doctype definition. Seems anytime I save a doctype this line gets removed.
